cloudflare_hostname_tls_setting: missing required hostname parameter

### Confirmation

- [x] This is a bug with an existing resource and is not a feature request or enhancement. Feature requests should be submitted with Cloudflare Support or your account team.
- [x] I have searched the issue tracker and my issue isn't already found.
- [x] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Hello
Thank you fix cloudflare_hostname_tls_setting valus, however I have new bug, optional parameter ```hostname (String) The hostname for which the tls settings are set. ``` is required when I apply resource.
With cURL ```hostname``` is a oprional and work correct.
```
curl --request PATCH \
"https://api.cloudflare.com/client/v4/zones/${zone_id}/settings/ciphers" \
--header "Content-Type: application/json" \
--header "Authorization: Bearer ${CLOUDFLARE_API_TOKEN}" \
--data '{"value": ["ECDHE-ECDSA-AES128-GCM-SHA256", "ECDHE-RSA-AES128-GCM-SHA256", "ECDHE-ECDSA-AES256-GCM-SHA384", "ECDHE-RSA-AES256-GCM-SHA384", "ECDHE-ECDSA-CHACHA20-POLY1305", "ECDHE-RSA-CHACHA20-POLY1305"]}' \
| jq

```

### Affected resource(s)

cloudflare_hostname_tls_setting

### Terraform configuration files

```hcl
terraform {
  required_providers {
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "~> 5.0"
    }
  }
  required_version = ">= 1.5.0"
}

resource "cloudflare_hostname_tls_setting" "it" {
  setting_id = "ciphers"
  value      = var.ciphers
  zone_id    = var.zone_id
}
variable "zone_id" {
  description = "CF zone id"
  type        = string
  default = "d1..."
}

variable "ciphers" {
  description = "The tls setting value."
  type        = list(string)
  # default = null
  default = ["ECDHE-ECDSA-AES128-GCM-SHA256", "ECDHE-RSA-AES128-GCM-SHA256", "ECDHE-ECDSA-AES256-GCM-SHA384", "ECDHE-RSA-AES256-GCM-SHA384", "ECDHE-ECDSA-CHACHA20-POLY1305", "ECDHE-RSA-CHACHA20-POLY1305"]
}
```

### Link to debug output

none

### Panic output

_No response_

### Expected output

smt like applied

### Actual output

Terraform will perform the following actions:

  # cloudflare_hostname_tls_setting.it will be created
  + resource "cloudflare_hostname_tls_setting" "it" {
      + created_at = (known after apply)
      + id         = (known after apply)
      + setting_id = "ciphers"
      + status     = (known after apply)
      + updated_at = (known after apply)
      + value      = [
          + "ECDHE-ECDSA-AES128-GCM-SHA256",
          + "ECDHE-RSA-AES128-GCM-SHA256",
          + "ECDHE-ECDSA-AES256-GCM-SHA384",
          + "ECDHE-RSA-AES256-GCM-SHA384",
          + "ECDHE-ECDSA-CHACHA20-POLY1305",
          + "ECDHE-RSA-CHACHA20-POLY1305",
        ]
      + zone_id    = "d1....."
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

cloudflare_hostname_tls_setting.it: Creating...
╷
│ Error: failed to make http request
│ 
│   with cloudflare_hostname_tls_setting.it,
│   on main.tf line 11, in resource "cloudflare_hostname_tls_setting" "it":
│   11: resource "cloudflare_hostname_tls_setting" "it" {
│ 
│ missing required hostname parameter


### Steps to reproduce

copy my code to file
set zone_id correct
terraform init
terraform apply

### Additional factoids

_No response_

### References

_No response_

```
 ✗ TF_LOG=DEBUG terraform apply                                 
2025-02-14T22:38:12.210Z [INFO]  Terraform version: 1.10.5
2025-02-14T22:38:12.210Z [DEBUG] using github.com/hashicorp/go-tfe v1.70.0
2025-02-14T22:38:12.210Z [DEBUG] using github.com/hashicorp/hcl/v2 v2.23.0
2025-02-14T22:38:12.210Z [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2025-02-14T22:38:12.210Z [DEBUG] using github.com/zclconf/go-cty v1.16.2
2025-02-14T22:38:12.210Z [INFO]  Go runtime version: go1.23.3
2025-02-14T22:38:12.210Z [INFO]  CLI args: []string{"terraform", "apply"}
2025-02-14T22:38:12.210Z [DEBUG] Attempting to open CLI config file: /home/qwerty/.terraformrc
2025-02-14T22:38:12.210Z [INFO]  Loading CLI configuration from /home/qwerty/.terraformrc
2025-02-14T22:38:12.210Z [DEBUG] checking for credentials in "/home/qwerty/.terraform.d/plugins"
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2025-02-14T22:38:12.210Z [DEBUG] will search for provider plugins in /home/qwerty/.terraform.d/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /home/qwerty/.local/share/terraform/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /usr/share/ubuntu/terraform/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /usr/share/gnome/terraform/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /home/qwerty/.local/share/flatpak/exports/share/terraform/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /var/lib/flatpak/exports/share/terraform/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2025-02-14T22:38:12.210Z [DEBUG] ignoring non-existing provider search directory /var/lib/snapd/desktop/terraform/plugins
2025-02-14T22:38:12.211Z [INFO]  Checkpoint disabled. Not running.
2025-02-14T22:38:12.211Z [INFO]  CLI command args: []string{"apply"}
2025-02-14T22:38:12.793Z [DEBUG] checking for provisioner in "."
2025-02-14T22:38:12.804Z [DEBUG] checking for provisioner in "/usr/bin"
2025-02-14T22:38:12.804Z [DEBUG] checking for provisioner in "/home/qwerty/.terraform.d/plugins"
2025-02-14T22:38:12.805Z [INFO]  backend/local: starting Apply operation
2025-02-14T22:38:12.808Z [DEBUG] created provider logger: level=debug
2025-02-14T22:38:12.808Z [INFO]  provider: configuring client automatic mTLS
2025-02-14T22:38:12.839Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 args=[".terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0"]
2025-02-14T22:38:12.840Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 pid=21735
2025-02-14T22:38:12.840Z [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0
2025-02-14T22:38:12.854Z [INFO]  provider.terraform-provider-cloudflare_v5.1.0: configuring server automatic mTLS: timestamp=2025-02-14T22:38:12.854Z
2025-02-14T22:38:12.874Z [DEBUG] provider: using plugin: version=6
2025-02-14T22:38:12.874Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: plugin address: address=/tmp/plugin1978285415 network=unix timestamp=2025-02-14T22:38:12.874Z
2025-02-14T22:38:13.015Z [DEBUG] No provider meta schema returned
2025-02-14T22:38:13.062Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-02-14T22:38:13.066Z [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 id=21735
2025-02-14T22:38:13.066Z [DEBUG] provider: plugin exited
2025-02-14T22:38:13.066Z [DEBUG] Building and walking validate graph
2025-02-14T22:38:13.067Z [DEBUG] ProviderTransformer: "cloudflare_hostname_tls_setting.it" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2025-02-14T22:38:13.067Z [DEBUG] ReferenceTransformer: "var.ciphers" references: []
2025-02-14T22:38:13.067Z [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" references: []
2025-02-14T22:38:13.067Z [DEBUG] ReferenceTransformer: "cloudflare_hostname_tls_setting.it" references: [var.ciphers var.zone_id]
2025-02-14T22:38:13.067Z [DEBUG] ReferenceTransformer: "var.zone_id" references: []
2025-02-14T22:38:13.067Z [DEBUG] ReferenceTransformer: "var.hostname" references: []
2025-02-14T22:38:13.067Z [DEBUG] Starting graph walk: walkValidate
2025-02-14T22:38:13.069Z [DEBUG] created provider logger: level=debug
2025-02-14T22:38:13.069Z [INFO]  provider: configuring client automatic mTLS
2025-02-14T22:38:13.077Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 args=[".terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0"]
2025-02-14T22:38:13.077Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 pid=21769
2025-02-14T22:38:13.077Z [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0
2025-02-14T22:38:13.093Z [INFO]  provider.terraform-provider-cloudflare_v5.1.0: configuring server automatic mTLS: timestamp=2025-02-14T22:38:13.093Z
2025-02-14T22:38:13.113Z [DEBUG] provider: using plugin: version=6
2025-02-14T22:38:13.113Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: plugin address: address=/tmp/plugin3002742258 network=unix timestamp=2025-02-14T22:38:13.112Z
2025-02-14T22:38:13.148Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-02-14T22:38:13.150Z [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 id=21769
2025-02-14T22:38:13.150Z [DEBUG] provider: plugin exited
2025-02-14T22:38:13.150Z [INFO]  backend/local: apply calling Plan
2025-02-14T22:38:13.150Z [DEBUG] Building and walking plan graph for NormalMode
2025-02-14T22:38:13.152Z [DEBUG] ProviderTransformer: "cloudflare_hostname_tls_setting.it (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2025-02-14T22:38:13.152Z [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" references: []
2025-02-14T22:38:13.152Z [DEBUG] ReferenceTransformer: "cloudflare_hostname_tls_setting.it (expand)" references: [var.ciphers var.zone_id]
2025-02-14T22:38:13.152Z [DEBUG] ReferenceTransformer: "var.zone_id" references: []
2025-02-14T22:38:13.152Z [DEBUG] ReferenceTransformer: "var.hostname" references: []
2025-02-14T22:38:13.152Z [DEBUG] ReferenceTransformer: "var.ciphers" references: []
2025-02-14T22:38:13.152Z [DEBUG] Starting graph walk: walkPlan
2025-02-14T22:38:13.153Z [DEBUG] created provider logger: level=debug
2025-02-14T22:38:13.153Z [INFO]  provider: configuring client automatic mTLS
2025-02-14T22:38:13.167Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 args=[".terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0"]
2025-02-14T22:38:13.167Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 pid=21783
2025-02-14T22:38:13.167Z [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0
2025-02-14T22:38:13.184Z [INFO]  provider.terraform-provider-cloudflare_v5.1.0: configuring server automatic mTLS: timestamp=2025-02-14T22:38:13.183Z
2025-02-14T22:38:13.202Z [DEBUG] provider: using plugin: version=6
2025-02-14T22:38:13.203Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: plugin address: address=/tmp/plugin972040641 network=unix timestamp=2025-02-14T22:38:13.202Z
2025-02-14T22:38:13.229Z [DEBUG] Resource instance state not found for node "cloudflare_hostname_tls_setting.it", instance cloudflare_hostname_tls_setting.it
2025-02-14T22:38:13.230Z [DEBUG] ReferenceTransformer: "cloudflare_hostname_tls_setting.it" references: []
2025-02-14T22:38:13.230Z [DEBUG] refresh: cloudflare_hostname_tls_setting.it: no state, so not refreshing
2025-02-14T22:38:13.248Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: Marking Computed attributes with null configuration values as unknown (known after apply) in the plan to prevent potential Terraform errors: @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:217 @module=sdk.framework tf_rpc=PlanResourceChange tf_resource_type=cloudflare_hostname_tls_setting tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=2f358ca0-4bcb-63bd-2174-f57eccedf011 timestamp=2025-02-14T22:38:13.246Z
2025-02-14T22:38:13.248Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: tf_req_id=2f358ca0-4bcb-63bd-2174-f57eccedf011 @module=sdk.framework tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_attribute_path="AttributeName(\"updated_at\")" tf_resource_type=cloudflare_hostname_tls_setting tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 timestamp=2025-02-14T22:38:13.246Z
2025-02-14T22:38:13.248Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: @module=sdk.framework tf_req_id=2f358ca0-4bcb-63bd-2174-f57eccedf011 tf_rpc=PlanResourceChange tf_attribute_path="AttributeName(\"created_at\")" tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_resource_type=cloudflare_hostname_tls_setting @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 timestamp=2025-02-14T22:38:13.247Z
2025-02-14T22:38:13.248Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: tf_attribute_path="AttributeName(\"id\")" tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 @module=sdk.framework tf_req_id=2f358ca0-4bcb-63bd-2174-f57eccedf011 tf_resource_type=cloudflare_hostname_tls_setting timestamp=2025-02-14T22:38:13.247Z
2025-02-14T22:38:13.248Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=2f358ca0-4bcb-63bd-2174-f57eccedf011 tf_attribute_path="AttributeName(\"status\")" tf_resource_type=cloudflare_hostname_tls_setting tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 @module=sdk.framework timestamp=2025-02-14T22:38:13.247Z
2025-02-14T22:38:13.251Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-02-14T22:38:13.254Z [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 id=21783
2025-02-14T22:38:13.254Z [DEBUG] provider: plugin exited
2025-02-14T22:38:13.255Z [DEBUG] building apply graph to check for errors
2025-02-14T22:38:13.255Z [DEBUG] Resource state not found for node "cloudflare_hostname_tls_setting.it", instance cloudflare_hostname_tls_setting.it
2025-02-14T22:38:13.255Z [DEBUG] ProviderTransformer: "cloudflare_hostname_tls_setting.it" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2025-02-14T22:38:13.255Z [DEBUG] ProviderTransformer: "cloudflare_hostname_tls_setting.it (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2025-02-14T22:38:13.255Z [DEBUG] ReferenceTransformer: "cloudflare_hostname_tls_setting.it" references: [var.ciphers var.zone_id]
2025-02-14T22:38:13.255Z [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" references: []
2025-02-14T22:38:13.255Z [DEBUG] ReferenceTransformer: "cloudflare_hostname_tls_setting.it (expand)" references: [var.ciphers var.zone_id]
2025-02-14T22:38:13.255Z [DEBUG] ReferenceTransformer: "var.zone_id" references: []
2025-02-14T22:38:13.255Z [DEBUG] ReferenceTransformer: "var.hostname" references: []
2025-02-14T22:38:13.255Z [DEBUG] ReferenceTransformer: "var.ciphers" references: []

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # cloudflare_hostname_tls_setting.it will be created
  + resource "cloudflare_hostname_tls_setting" "it" {
      + created_at = (known after apply)
      + id         = (known after apply)
      + setting_id = "ciphers"
      + status     = (known after apply)
      + updated_at = (known after apply)
      + value      = [
          + "ECDHE-ECDSA-AES128-GCM-SHA256",
          + "ECDHE-RSA-AES128-GCM-SHA256",
          + "ECDHE-ECDSA-AES256-GCM-SHA384",
          + "ECDHE-RSA-AES256-GCM-SHA384",
          + "ECDHE-ECDSA-CHACHA20-POLY1305",
          + "ECDHE-RSA-CHACHA20-POLY1305",
        ]
      + zone_id    = "d19..."
    }

Plan: 1 to add, 0 to change, 0 to destroy.
2025-02-14T22:38:13.261Z [DEBUG] command: asking for input: "\nDo you want to perform these actions?"

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

2025-02-14T22:38:17.904Z [INFO]  backend/local: apply calling Apply
2025-02-14T22:38:17.904Z [DEBUG] Building and walking apply graph for NormalMode plan
2025-02-14T22:38:17.904Z [DEBUG] Resource state not found for node "cloudflare_hostname_tls_setting.it", instance cloudflare_hostname_tls_setting.it
2025-02-14T22:38:17.905Z [DEBUG] ProviderTransformer: "cloudflare_hostname_tls_setting.it" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2025-02-14T22:38:17.905Z [DEBUG] ProviderTransformer: "cloudflare_hostname_tls_setting.it (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/cloudflare/cloudflare"]
2025-02-14T22:38:17.905Z [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/cloudflare/cloudflare\"]" references: []
2025-02-14T22:38:17.905Z [DEBUG] ReferenceTransformer: "cloudflare_hostname_tls_setting.it (expand)" references: [var.ciphers var.zone_id]
2025-02-14T22:38:17.905Z [DEBUG] ReferenceTransformer: "var.zone_id" references: []
2025-02-14T22:38:17.905Z [DEBUG] ReferenceTransformer: "var.hostname" references: []
2025-02-14T22:38:17.905Z [DEBUG] ReferenceTransformer: "var.ciphers" references: []
2025-02-14T22:38:17.905Z [DEBUG] ReferenceTransformer: "cloudflare_hostname_tls_setting.it" references: [var.ciphers var.zone_id]
2025-02-14T22:38:17.906Z [DEBUG] Starting graph walk: walkApply
2025-02-14T22:38:17.906Z [DEBUG] created provider logger: level=debug
2025-02-14T22:38:17.906Z [INFO]  provider: configuring client automatic mTLS
2025-02-14T22:38:17.920Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 args=[".terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0"]
2025-02-14T22:38:17.921Z [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 pid=21855
2025-02-14T22:38:17.921Z [DEBUG] provider: waiting for RPC address: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0
2025-02-14T22:38:17.933Z [INFO]  provider.terraform-provider-cloudflare_v5.1.0: configuring server automatic mTLS: timestamp=2025-02-14T22:38:17.933Z
2025-02-14T22:38:17.952Z [DEBUG] provider: using plugin: version=6
2025-02-14T22:38:17.953Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: plugin address: address=/tmp/plugin383397761 network=unix timestamp=2025-02-14T22:38:17.952Z
2025-02-14T22:38:17.989Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: Marking Computed attributes with null configuration values as unknown (known after apply) in the plan to prevent potential Terraform errors: tf_req_id=69675dba-b788-245e-8f9a-be2b40c9dd2e tf_resource_type=cloudflare_hostname_tls_setting @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:217 @module=sdk.framework tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_rpc=PlanResourceChange timestamp=2025-02-14T22:38:17.989Z
2025-02-14T22:38:17.990Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: tf_attribute_path="AttributeName(\"id\")" tf_req_id=69675dba-b788-245e-8f9a-be2b40c9dd2e tf_resource_type=cloudflare_hostname_tls_setting @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 @module=sdk.framework tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_rpc=PlanResourceChange timestamp=2025-02-14T22:38:17.989Z
2025-02-14T22:38:17.990Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: tf_rpc=PlanResourceChange @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 @module=sdk.framework tf_attribute_path="AttributeName(\"status\")" tf_resource_type=cloudflare_hostname_tls_setting tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=69675dba-b788-245e-8f9a-be2b40c9dd2e timestamp=2025-02-14T22:38:17.990Z
2025-02-14T22:38:17.990Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 tf_attribute_path="AttributeName(\"updated_at\")" tf_provider_addr=registry.terraform.io/cloudflare/cloudflare @module=sdk.framework tf_req_id=69675dba-b788-245e-8f9a-be2b40c9dd2e tf_resource_type=cloudflare_hostname_tls_setting tf_rpc=PlanResourceChange timestamp=2025-02-14T22:38:17.990Z
2025-02-14T22:38:17.990Z [DEBUG] provider.terraform-provider-cloudflare_v5.1.0: marking computed attribute that is null in the config as unknown: tf_attribute_path="AttributeName(\"created_at\")" tf_req_id=69675dba-b788-245e-8f9a-be2b40c9dd2e tf_rpc=PlanResourceChange @module=sdk.framework tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_resource_type=cloudflare_hostname_tls_setting @caller=github.com/hashicorp/terraform-plugin-framework@v1.13.0/internal/fwserver/server_planresourcechange.go:467 timestamp=2025-02-14T22:38:17.990Z
cloudflare_hostname_tls_setting.it: Creating...
2025-02-14T22:38:17.991Z [INFO]  Starting apply for cloudflare_hostname_tls_setting.it
2025-02-14T22:38:17.992Z [DEBUG] cloudflare_hostname_tls_setting.it: applying the planned Create change
2025-02-14T22:38:17.994Z [ERROR] provider.terraform-provider-cloudflare_v5.1.0: Response contains error diagnostic: @module=sdk.proto diagnostic_detail="missing required hostname parameter" diagnostic_severity=ERROR diagnostic_summary="failed to make http request" tf_proto_version=6.7 @caller=github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov6/internal/diag/diagnostics.go:58 tf_provider_addr=registry.terraform.io/cloudflare/cloudflare tf_req_id=63a7819f-fca9-2e00-93f9-f96e5de78227 tf_resource_type=cloudflare_hostname_tls_setting tf_rpc=ApplyResourceChange timestamp=2025-02-14T22:38:17.994Z
2025-02-14T22:38:17.995Z [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2025-02-14T22:38:17.995Z [ERROR] vertex "cloudflare_hostname_tls_setting.it" error: failed to make http request
╷
│ Error: failed to make http request
│ 
│   with cloudflare_hostname_tls_setting.it,
│   on main.tf line 11, in resource "cloudflare_hostname_tls_setting" "it":
│   11: resource "cloudflare_hostname_tls_setting" "it" {
│ 
│ missing required hostname parameter
╵
2025-02-14T22:38:17.997Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2025-02-14T22:38:17.999Z [INFO]  provider: plugin process exited: plugin=.terraform/providers/registry.terraform.io/cloudflare/cloudflare/5.1.0/linux_amd64/terraform-provider-cloudflare_v5.1.0 id=21855
2025-02-14T22:38:17.999Z [DEBUG] provider: plugin exited

```

--- a/internal/services/hostname_tls_setting/model.go
+++ b/internal/services/hostname_tls_setting/model.go
@@ -16,7 +16,7 @@ type HostnameTLSSettingModel struct {
 	ID        types.String      `tfsdk:"id" json:"-,computed"`
 	SettingID types.String      `tfsdk:"setting_id" path:"setting_id,required"`
 	ZoneID    types.String      `tfsdk:"zone_id" path:"zone_id,required"`
-	Hostname  types.String      `tfsdk:"hostname" path:"hostname,optional"`
+	Hostname  types.String      `tfsdk:"hostname" path:"hostname,required"`
 	Value     types.Dynamic     `tfsdk:"value" json:"value,required"`
 	CreatedAt timetypes.RFC3339 `tfsdk:"created_at" json:"created_at,computed" format:"date-time"`
 	Status    types.String      `tfsdk:"status" json:"status,computed"`

--- a/internal/services/hostname_tls_setting/schema.go
+++ b/internal/services/hostname_tls_setting/schema.go
@@ -50,7 +50,7 @@ func ResourceSchema(ctx context.Context) schema.Schema {
 			},
 			"hostname": schema.StringAttribute{
 				Description:   "The hostname for which the tls settings are set.",
-				Optional:      true,
+				Required:      true,
 				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
 			},
 			"value": schema.DynamicAttribute{

